src/main/java/org/testng/SuiteRunner.java

```diff
@@ -161,8 +161,10 @@ public int compare(XmlTest arg0, XmlTest arg1) {
       //
       // Install the method interceptor, if any was passed
       //
-      if (m_methodInterceptor != null) {
-        tr.setMethodInterceptor(m_methodInterceptor);
+      if (!m_methodInterceptors.isEmpty()) {
```

Comment:
I'm not sure this condition is helpful because iterators work with empty lists.
But you may check to null as someone could send null (even if I think that create a new list if null will be better).